test/main/store/migrations/index.test.js

```diff
@@ -528,3 +528,166 @@ describe('migration 20', () => {
+    expect(sepolia.connection.secondary.current).toBe('custom')
+    expect(sepolia.on).toBe(false)
+  })
+
+  it('does not change existing Sepolia network information', () => {
+    state.main.networks.ethereum[11155111] = {
+      explorer: 'https://custom-explorer.sepolia.dev',
+      connection: {
+        primary: { on: true, current: 'local' }
+      }
+    }
+
+    const updatedState = migrations.apply(state, 21)
+    const sepolia = updatedState.main.networks.ethereum[11155111]
+
+    expect(sepolia.explorer).toBe('https://custom-explorer.sepolia.dev')
+    expect(sepolia.connection.primary.on).toBe(true)
+    expect(sepolia.connection.primary.current).toBe('local')
+  })
+
+  it('adds Sepolia network meta information when none exists', () => {
+    delete state.main.networksMeta.ethereum[11155111]
+
+    const updatedState = migrations.apply(state, 21)
+    const sepolia = updatedState.main.networksMeta.ethereum[11155111]
+
+    expect(sepolia.gas.fees.maxFeePerGas).toBe(undefined)
+    expect(sepolia).toMatchObject({
+      gas: { fees: {} , price: { selected: 'standard', levels: {} } }
+    })
+  })
+
+  it('does not change existing Sepolia meta network information', () => {
+    state.main.networksMeta.ethereum[11155111] = {
+      gas: {
+        fees: {
+          maxFeePerGas: '0xf'
+        }
+      }
+    }
+
+    const updatedState = migrations.apply(state, 21)
+    const sepolia = updatedState.main.networksMeta.ethereum[11155111]
+
+    expect(sepolia.gas.fees.maxFeePerGas).toBe('0xf')
+  })
+
+  it('adds Sepolia network preset information when none exists', () => {
+    delete state.main.networkPresets.ethereum[11155111]
+
+    const updatedState = migrations.apply(state, 21)
+    const sepolia = updatedState.main.networkPresets.ethereum[11155111]
+
+    expect(sepolia).toMatchObject({
+      infura: 'infuraSepolia'
+    })
+  })
+
+  const removedGoerliRPCs = ['mudit', 'slockit', 'prylabs']
+
+  removedGoerliRPCs.forEach((removedRPCName) => {
+    it(`resets and turns off goerli when the ${removedRPCName} RPC is active as a primary connection`, () => {
+      state.main.networks.ethereum[5].connection.primary = { 
+        on: true, 
+        current: removedRPCName, 
+        status: 'disconnected', 
+        connected: false, 
+        type: '', 
+        network: '', 
+        custom: '' 
+      }
+
+      const updatedState = migrations.apply(state, 21)
+      const goerli = updatedState.main.networks.ethereum[5]
+
+      expect(goerli.connection).toMatchObject({
+        primary: { on: true, current: 'infura', status: 'loading', connected: false, type: '', network: '', custom: '' },
+        secondary: { on: false, current: 'custom', status: 'loading', connected: false, type: '', network: '', custom: '' }
+      })
+      expect(goerli.on).toBe(false)
+    })
+
+    it(`resets and turns off goerli when the ${removedRPCName} RPC is active as a secondary connection`, () => {
```

Comment:
as mentioned above I don't think this is the right behavior. probably we should just turn off this connection if its secondary

Reply:
On reflection your suggestion is heading in the right direction, but I don't think it is the right behaviour.  We need to reset the current value where the current value is a removed RPC, regardless of whether it is primary or secondary, active or not.  We should turn off Goerli completely if the above process results in an active (not just enabled) sub-connection being removed.